### PR TITLE
New pages for profile, new components and styles

### PR DESCRIPTION
**Pull request by tomrndom**
WIP

ref: https://tipit.avaza.com/task#!sortby=DateUpdatedDesc&task=2877727

Signed-off-by: Tomás Castillo <tcastilloboireau@gmail.com>

### CODE REVIEW

File: src/actions/user-actions.js [hunk 1]
```diff
@@ -327,6 +336,10 @@ export const updateMembershipType = (type) => (dispatch, getState) => {
     authErrorHandler
   )(params)(dispatch)
     .then((payload) => {
+      if (currentMembershipType === MEMBERSHIP_TYPE_NONE) {
+        dispatch(updateProfile(profile));
+        dispatch(updateIDPProfile(idpProfile));
+      }
       dispatch(stopLoading());
```
Comment (smarcet) [inline src/actions/user-actions.js:343]: add a visual feedback for a successful operation 
a modal message would do the trick

File: src/components/ProfileManagementComponent.js [hunk 2]
```diff
@@ -188,6 +193,55 @@ export const ProfileManagement = ({
     }
   };
 
+  const handleSubmitApplication = () => {
+    if (!publicInformation.firstName || !publicInformation.lastName || !publicInformation.identifier || !publicInformation.email) {
+      const msg = `Required field missing`;
+      Swal.fire("Validation error", msg, "warning");
```
Comment (smarcet) [inline src/components/ProfileManagementComponent.js:199]: here you could use.a return and remove the else branch

File: src/components/Affiliations.js [hunk 3]
```diff
@@ -89,10 +89,14 @@ const Affiliations = class extends React.Component {
...
                                     <td key="end_date">{formatEpoch(a.end_date, "YYYY-MM-DD")}</td>
                                     <td key="is_current">{a.is_current ? 'Yes' : 'No'}</td>
-                                    <td>
-                                        <button data-id={a.id} onClick={this.onHandleEdit}>Edit</button>
+                                    <td style={{display: 'flex'}}>
+                                        <button data-id={a.id} onClick={this.onHandleEdit}>
+                                            <i className={`fa fa-pencil`} />
+                                        </button>
                                         &nbsp;
-                                        <button data-id={a.id} onClick={this.onHandleDelete}>Delete</button>
```
Comment (smarcet) [inline src/components/Affiliations.js:95]: on handleDelete use ev.currentTarget instead of ev.target

File: src/components/Affiliations.js [hunk 4]
```diff
@@ -89,10 +89,14 @@ const Affiliations = class extends React.Component {
                                     <td key="start_date">{formatEpoch(a.start_date, "YYYY-MM-DD")}</td>
                                     <td key="end_date">{formatEpoch(a.end_date, "YYYY-MM-DD")}</td>
                                     <td key="is_current">{a.is_current ? 'Yes' : 'No'}</td>
-                                    <td>
-                                        <button data-id={a.id} onClick={this.onHandleEdit}>Edit</button>
+                                    <td style={{display: 'flex'}}>
+                                        <button data-id={a.id} onClick={this.onHandleEdit}>
```
Comment (smarcet) [inline src/components/Affiliations.js:93]: on onHandleEdit use ev.currentTarget instead of ev.target

**Review by smarcet — CHANGES_REQUESTED** [review]
@tomrndom plese review comments

**Comment by iamweswilson** [comment]
Update this text to the following: 

"Speakers agree that OpenInfra Foundation may record and publish any talks presented during the OpenInfra Summit. If you submit a proposal on behalf of a speaker, you represent to OpenInfra Foundation that you have the authority to submit the proposal on the speaker's behalf and agree to the recording and publication of their presentation."

screenshot


<details><summary>Browser metadata</summary>





```
Path:      /a/profile/speaker
Browser:   Safari 15.4 on Mac OS 10.15.7
Viewport:  1440 x 725 @2x
Language:  en-US
Cookies:   Disabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-254--openinfra-dev.netlify.app%2Fa%2Fprofile%2Fspeaker&os=OS%20X&os_version=Catalina&browser=Safari&browser_version=15.0&resolution=1440x725&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-254--openinfra-dev.netlify.app/a/profile/speaker) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6266eb85666b630009054af8&commentID=6267084dbc8c460b7418425a&resolution=resolved)

**Comment by iamweswilson** [comment]
Can we make sure the pics shown are circles? I think this is happening because this pic width is not 200px

screenshot


<details><summary>Browser metadata</summary>





```
Path:      /a/profile/speaker
Browser:   Safari 15.4 on Mac OS 10.15.7
Viewport:  1440 x 725 @2x
Language:  en-US
Cookies:   Disabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-254--openinfra-dev.netlify.app%2Fa%2Fprofile%2Fspeaker&os=OS%20X&os_version=Catalina&browser=Safari&browser_version=15.0&resolution=1440x725&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-254--openinfra-dev.netlify.app/a/profile/speaker) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6266eb85666b630009054af8&commentID=626709c89197160d43acd805&resolution=resolved)

**Comment by iamweswilson** [comment]
'&' converts to unicode when saved, can we keep what's typed?

screenshot


<details><summary>Browser metadata</summary>





```
Path:      /a/profile/speaker
Browser:   Safari 15.4 on Mac OS 10.15.7
Viewport:  1440 x 725 @2x
Language:  en-US
Cookies:   Disabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-254--openinfra-dev.netlify.app%2Fa%2Fprofile%2Fspeaker&os=OS%20X&os_version=Catalina&browser=Safari&browser_version=15.0&resolution=1440x725&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-254--openinfra-dev.netlify.app/a/profile/speaker) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6266eb85666b630009054af8&commentID=62670ac693799f08b5db09c1&resolution=resolved)

**Comment by smarcet** [comment]
> I notice we're using a different data source for https://deploy-preview-254--openinfra-dev.netlify.app/a/profile/speaker and https://deploy-preview-254--openinfra-dev.netlify.app/a/profile. Is it possible to consolidate those or have we kept it this way to allow speakers to change the image they use for their sessions?
> 
> Browser metadata
> [Open Deploy Preview](https://deploy-preview-254--openinfra-dev.netlify.app/a/profile/speaker) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6266eb85666b630009054af8&commentID=62670bd6f02f5405ce33ad5b&resolution=resolved)

@iamweswilson that is done on that way to keep data for speaker separated from the member its has been on that way since forever and change it will entail a massive change 
regards

**Comment by smarcet** [comment]
> Can we only show this button when nominations are open? Doesn't look like we're currently showing it on https://www.openstack.org/profile/election
> 
> screenshot
> 
> Browser metadata
> [Open Deploy Preview](https://deploy-preview-254--openinfra-dev.netlify.app/a/profile/elections) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6266eb85666b630009054af8&commentID=62670b57301d3c049a20a027&resolution=resolved)

@iamweswilson this is pointing to devbranch.openstack.org 
could you check there ?
regards

**Review by smarcet — APPROVED** [review]
LGTM

**Comment by iamweswilson** [comment]
I don't see the button on devbranch either image

**Pull request merged**